### DISCUSSION TITLE
Permission downloads.open Should be an Optional Permission

### DISCUSSION THREAD

**Discussion by smashedr** [aclap-dev/video-downloadhelper#597] [Help and Q&A] [answered]
### Your question

The permission `downloads.open` should never be granted to any web extension because it allows malware to be executed on the host system. It should be implemented as an `optional_permission`. Otherwise, many people will not use this extension anymore going forward. 

https://developer.chrome.com/docs/extensions/reference/api/permissions
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/optional_permissions

### Addon & CoApp version and details

This is not a client issue.

**Comment by paulrouget** [comment]
I think that's very fair. Gonna do that. Thank you!

**Reply by paulrouget** [reply, marked as answer]
Coming up in version 9.0.1.21.

**Comment by smashedr** [comment]
Thanks for addressing this issue. I got the automatic update today. Very pleased with the response time.

While just curious, I was unable to find where to request the extra permissions. As an extension developer myself I wanted to see how you implemented the permissions request.

Also, possibly unrelated, but when the update was installed, my browser took me to this page: https://www.downloadhelper.net/changelog/mozilla/stable/ which at the time of writing this, shows "Page not found". Just wanted to let you know in case it was a bug.

My debugging info if it is needed:
```
version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:126.0) Gecko/20100101 Firefox/126.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "download_directory": "C:\\Users\\Shane\\Downloads",
    "concurrent_downloads_max": 6,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "show_context_menu": false,
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 5
}
```

Thanks again for the update, very pleased that I don't have to uninstall this extension, as its is one of the best.

**Reply by GrampaWildWilly** [reply]
That non-functioning URL you mentioned is functioning now.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.

As an extension developer, surely you know about about:addons.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/6243955e-230f-4792-ae70-da27b9014611)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c0fa49fc-a589-4487-b944-fbb422b172b7)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/756d9bac-2cde-4957-8c63-4986f1535d27)

**Reply by smashedr** [reply]
Not sure what any of this has to do with my question about implementing the feature on Chrome https://github.com/aclap-dev/video-downloadhelper/discussions/597#discussioncomment-9779096

**Comment by smashedr** [comment]
Is this update not going to be implement in the Chrome version?
